### PR TITLE
Move common testing methods to the separate package

### PR DESCRIPTION
**Pull request by cynepco3hahue**
- move methods from the machine health check tests
- move methods from the machine disruption tests

### CODE REVIEW

File: pkg/controller/disruption/disruption_test.go [hunk 1]
```diff
@@ -492,21 +396,32 @@ func TestBuildDisruptedMachineMap(t *testing.T) {
...
-	r := newFakeReconciler(nil, healthyNode, unhealthyNode)
+	// machine disrupted before the timeout
+	nodeDisruptedBeforeTimeout := maotesting.NewNode("nodeDisruptedBeforeTimeout", true)
+	disruptedMachineBeforeTimeout := maotesting.NewMachine("disruptedMachineBeforeTimeout", nodeDisruptedBeforeTimeout.Name)
+
+	// machine disrupted after the timeout
+	nodeDisruptedAfterTimeout := maotesting.NewNode("nodeDisruptedAfterTimeout", true)
+	disruptedMachineAfterTimeout := maotesting.NewMachine("disruptedMachineAfterTimeout", nodeDisruptedAfterTimeout.Name)
+
+	r := newFakeReconciler(nil, healthyNode, unhealthyNode, deletedNode, nodeDisruptedBeforeTimeout, nodeDisruptedAfterTimeout)
```
Comment (ingvagabund) [inline pkg/controller/disruption/disruption_test.go]: Does not look like just moving the code from from place to another. Before the change `newFakeReconciler` accepted three arguments. Now, it accepts 6. Please, separate this change into another commit for better readability.

File: pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go [hunk 2]
```diff
@@ -223,102 +115,173 @@ func TestGetNodeCondition(t *testing.T) {
...
-	fakeMachine.Status = mapiv1alpha1.MachineStatus{
-		NodeRef: &corev1.ObjectReference{
-			Namespace: "",
-			Name:      "healthy",
-		},
+func testReconcile(t *testing.T, remediationWaitTime time.Duration, initObjects ...runtime.Object) {
+	// healthy node
+	nodeHealthy := maotesting.NewNode("healthy", true)
+	nodeHealthy.Annotations = map[string]string{
+		machineAnnotationKey: fmt.Sprintf("%s/%s", namespace, "machineWithNodehealthy"),
```
Comment (ingvagabund) [inline pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go]: Here, as well. This extends the original code. Please, put it into another file.

Comment (cynepco3hahue) [reply pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go]: done

File: pkg/controller/disruption/disruption_test.go (hunk 1 above)
Comment (cynepco3hahue) [reply pkg/controller/disruption/disruption_test.go]: done

**Comment by cynepco3hahue** [comment]
@ingvagabund Can you please check again?

File: pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go (hunk 2 above)
Comment (beekhof) [reply pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go]: missing a push?  i don't see a new file

**Comment by cynepco3hahue** [comment]
this one is new one https://github.com/openshift/machine-api-operator/pull/327/files#diff-a1352506d13977f468ac1dd702ac3f7c

File: pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go (hunk 2 above)
Comment (ingvagabund) [reply pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go]: Oh :) sorry, I meant commit, not file :-|

Comment (cynepco3hahue) [reply pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go]: @ingvagabund yep I understood you :smile: I separated it into two commits, can you please check it?
